tests/playwright/src/sso-extension.spec.ts

```diff
@@ -0,0 +1,160 @@
+  page = await pdRunner.start();
+  pdRunner.setVideoAndTraceName('sso-e2e');
+
+  const welcomePage = new WelcomePage(page);
+  await welcomePage.handleWelcomePage(true);
+  navBar = new NavigationBar(page);
+});
+
+afterAll(async () => {
+  await pdRunner.close();
+});
+
+describe('Red Hat Authentication extension verification', async () => {
+
+  test('Go to settings and check if extension is already installed', async () => {
+    const settingsBar = await navBar.openSettings();
+    const extensions = await settingsBar.getCurrentExtensions();
+    if (await extensionExists(extensions, extensionName)) {
+      extensionInstalled = true;
+    }
+  });
+
+  test.runIf(extensionInstalled)(
+    'Uninstalled previous version of sso extension',
+    async () => {
+      await removeExtension(extensionName);
+    },
+    60000,
+  );
+
+  test('Extension can be installed using OCI image', async () => {
+    const settingsBar = await navBar.openSettings();
+    const settingsExtensionPage = await settingsBar.openTabPage(SettingsExtensionsPage);
+    await playExpect(settingsExtensionPage.heading).toBeVisible();
+    await settingsExtensionPage.installExtensionFromOCIImage(imageName);
+  }, 200000);
+
+  test('Extension record is added under installed extensions', async () => {
+    const settingsBar = await navBar.openSettings();
+    const settingsExtensionPage = await settingsBar.openTabPage(SettingsExtensionsPage);
+    const extension = settingsExtensionPage.getExtensionRowFromTable(extensionLabelName);
+    await playExpect(extension).toBeVisible();
+    await extension.scrollIntoViewIfNeeded();
+    const extensionDetails = extension.getByRole('cell', { name: 'Extension Details' });
+    const details = extensionDetails.getByLabel('Connection Status Label');
+    await playExpect(details).toHaveText('RUNNING');
+  });
+
+  test('Extension appears under Settings bar extensions', async () => {
+    const settingsBar = await navBar.openSettings();
+    const extensions = await settingsBar.getCurrentExtensions();
+    await playExpect.poll(async () => await extensionExists(extensions, extensionName), { timeout: 30000 }).toBeTruthy();
+  });
+
+  test('SSO provider is available in Authentication Page', async (context) => {
+    console.log(`Running Test: ${context.task.name}`);
+    console.log(`Opening settings`);
+    const settingsBar = await navBar.openSettings();
+    const authPage = await settingsBar.openTabPage(AuthenticationPage);
+    await playExpect(authPage.heading).toHaveText('Authentication');
+    const provider = authPage.getProvider(authProviderName);
+    await playExpect(provider.getByLabel('Provider Information').getByLabel('Provider Name')).toHaveText(authProviderName);
+    await playExpect(provider.getByLabel('Provider Information').getByLabel('Provider Status')).toHaveText('Logged out');
+    await playExpect(provider.getByLabel('Provider Actions').getByRole('button')).toContainText('Sign in');
```

Comment:
The provider fields from this and the previous two lines should be accessible directly from the authentication POM, it shouldn't be needed to handle the labels at test level like this.

Reply:
I have not yet seen other examples how providers would look like, to is it not in authenticationPage, yet. Or you want me to skip that one level or locator findings: getByLabel(Provider Inromation/Actions) ?

Reply:
I would think that the provider name and status is be should be exposed from the POM in some way

Reply:
These fields with particular provider do not exist before you actually install some providers. I could create an issue to add some getter based on the name of the provider, since they do not have fixed names.

Reply:
Fair enough